index.xml

```diff
@@ -3948,6 +3967,101 @@ setlike&lt;<i>type</i>&gt;;</pre>
+
+          <p>
+            A <dfn data-export='' id='dfn-namespace'>namespace</dfn> is a definition (matching <a
+            class='sym' href='#prod-Namespace'>Namespace</a>) that declares a global singleton with
+            associated behaviors.
+          </p>
+
+          <pre class='syntax'>namespace <i>identifier</i> {
+  <i>namespace-members…</i>
+};</pre>
+
+          <p>
+            A namespace is a specification of a set of <dfn data-export='' data-lt='namespace
+            member' id='dfn-namespace-member'>namespace members</dfn> (matching <a class='sym'
+            href='#prod-NamespaceMembers'>NamespaceMembers</a>), which are the <a class='dfnref'
+            href='#dfn-regular-operation'>regular operations</a> that appear between the braces in
+            the namespace declaration. These operations describe the behaviors packaged into the
+            namespace.
+          </p>
+
+          <p>
+            As with interfaces, the IDL for namespaces can be split into multiple parts by using
+            <dfn data-export='' id='dfn-partial-namespace'>partial namespace</dfn> definitions
+            (matching <span class='sym'>"partial"</span> <a class='sym'
+            href='#prod-Namespace'>Namespace</a>). The <a class='dfnref'
+            href='#dfn-identifier'>identifier</a> of a partial namespace definition <span
+            class='rfc2119'>MUST</span> be the same as the identifier of a namespace definition.
+            All of the members that appear on each of the partial namespace definitions are
+            considered to be members of the namespace itself.
+          </p>
+
+          <pre class='syntax'>namespace <em>SomeNamespace</em> {
+  <i>namespace-members…</i>
+};
+
+partial namespace <em>SomeNamespace</em> {
+  <i>namespace-members…</i>
+};</pre>
+
+          <div class='note'>
+            <p>
+              As with partial interface definitions, partial namespace definitions are intended for
+              use as a specification editorial aide, allowing the definition of an interface to be
+              separated over more than one section of the document, and sometimes multiple
+              documents.
+            </p>
+          </div>
+
+          <p>
+            The order that members appear has significance both for <a
+            href='#idl-overloading'>overloading</a> and for property enumeration in the <a
+            href='#es-namespaces'>ECMAScript binding</a>.
+          </p>
+
+          <p>
+            Note that unlike interfaces or dictionaries, namespaces do not create types.
+          </p>
+
+          <p>
+            Only the <a class='xattr' href='#Exposed'>[Exposed]</a> and <a class='xattr'
+            href='#SecureContext'>[SecureContext]</a> extended attributes are applicable to
+            namespaces. When declaring a <a class='dfnref' href='#dfn-partial-namespace'>partial
+            namespace</a>, the partial namespace must have the same extended attributes (and
```

Comment:
Why?  This seems like an overly onerous requirement to me.  In particular, why would we prohibit this:

    [Exposed=(Window,Worker)]
    namespace foo {
      void func1();
    };
    [Exposed=Worker]
    partial namespace foo {
      void func2();
    };

as a mechanism for extending a namespace with some members that have more restricted exposure?  Similar for [SecureContext].  I don't see a reason to make these extended attributes work differently from the way they work for interfaces, if we're going to support partial namespaces at all.